legend-engine-xts-data-space/legend-engine-xt-data-space-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/DataSpaceCompilerExtension.java

```diff
@@ -160,10 +143,17 @@ public Iterable<? extends Processor<?>> getExtraProcessors()
 
                     // executables
                     metamodel._executables(dataSpace.executables != null ? ListIterate.collect(dataSpace.executables, executable ->
-                            new Root_meta_pure_metamodel_dataSpace_DataSpaceExecutable_Impl("", null, context.pureModel.getClass("meta::pure::metamodel::dataSpace::DataSpaceExecutable"))
-                                    ._title(executable.title)
-                                    ._description(executable.description)
-                                    ._executable(context.pureModel.getPackageableElement(executable.executable.path, executable.executable.sourceInformation))) : null);
+                            executable instanceof DataSpacePackageableElementExecutable ?
```

Comment:
can we rewrite this using if statements to make it more readonly.
`if (x instanceof)`
then also throw if there is handler for it.